src/devices/GoPiGo3/GoPiGo3.cs

```diff
@@ -0,0 +1,887 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.GoPiGo3.Models;
+using System;
+using System.Device.Spi;
+using System.Device.Spi.Drivers;
+using System.Diagnostics;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Iot.Device.GoPiGo3
+{
+    /// <summary>
+    /// A GoPiGo3 class for a GoPiGo3 Dexter Industries robot kit
+    /// </summary>
+    public class GoPiGo : IDisposable
+    {
+        private const int DefaultMotorGearRatio = 120;
+        private const int DefaultEncoderTicksPerRotation = 6;
+        private const int GrooveI2cLengthLimit = 32;
+        private const string ManufacturerName = "Dexter Industries";
+        private const string BoardName = "GoPiGo3";
+        private static SpiDevice _spiDevice = null;
```

Comment:
We are going to be introducing some changes with the way our API works and add support for devices that can provide Spi devices. For this reason, we have been pushing for bindings like this to have a spidevice passed into the constructor, instead of you constructing it inside your binding. Because of this, I would advice you change your constructor to instead look like:

```c#
public GoPiGo(SpiDevice spiDevice, bool autoDetect = true);
```

That way you will push the work of initializing to the consumer, and allow support for extra devices, and also, you will be able to change this member to not be static any longer.

Reply:
ok, I'll adjust it that way